caliban/platform/gke/cluster.py

```diff
@@ -188,13 +188,13 @@ class Cluster(object):
   """
 
   # --------------------------------------------------------------------------
-  def __init__(self, name: str, project_id: str, zone: str,
+  def __init__(self, name: Optional[str], project_id: str, zone: str,
                credentials: Credentials):
     self._cluster_client = None
-    self._gke_cluster = None
-    self._core_api = None
-    self._batch_api = None
-    self._apps_api = None
+    self._gke_cluster: Optional[GKECluster] = None
```

Comment:
nice, didn't realize you could type variables inline.

Reply:
now that we require python >= 3.6 we can!